### PR TITLE
Stack

### PR DESCRIPTION
**Pull request by george-e-shaw-iv**
* Implemented the stack data structure.

### CODE REVIEW

**Comment by brayo-pip** [comment]
@cclauss this is more ur domain

File: data-structures/stack/stack.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
+// Package stack implements the Stack data structure, LIFO.
+package stack
+
+// Stack is the type that implements the Stack data structure.
+type Stack struct {
```
Comment (Mystigan) [inline data-structures/stack/stack.go:5]: Why enclose it in a struct?
You could try something like `type Stack []interface{}`

**Review by Mystigan — CHANGES_REQUESTED** [review]
- Keep the stack type unexported and add an exported func `InitStack` or `New` to initialize an empty stack. This would ensure that a user can import and use this library, at the same time not modify the stack instance directly.
- Add some tests.

**Comment by tjgurwara99** [comment]
Please take a look at our new [contribution guidelines](https://github.com/TheAlgorithms/Go/blob/d2f12849df76203d80762ceca5105531d3424384/CONTRIBUTING.md) and make changes accordingly. Currently this PR is not up to standard. If no changes are made in the next two weeks, I will consider this PR to be stale and will close this PR then.

**Pull request closed without merging**